R/create-population.R

```diff
@@ -0,0 +1,114 @@
+#' @param gestationalAgeMin min gestational age for the population (optional, for human species only)
+#' @param gestationalAgeMax max gestational age for the population (optional, for human species only)
+#' @param gestationalAgeUnit Unit in which the gestational age value is defined. Default is kg/m²
+#' @param moleculeOntogenies Optional list of \code{MoleculeOntogeny} that will be used to retrieve ontogeny information for molecules.
+#'
+#' @return An instance of \code{PopulationCharacteristics} to be used in conjunction with \code{createPopulation}
+#'
+#' @export
+createPopulationCharacteristics <- function(
+                                            species,
+                                            population = NULL,
+                                            numberOfIndividuals,
+                                            proportionOfFemales = 50,
+                                            weightMin = NULL,
+                                            weightMax = NULL,
+                                            weightUnit = "kg",
+                                            heightMin = NULL,
+                                            heightMax = NULL,
+                                            heightUnit = "cm",
+                                            ageMin = NULL,
+                                            ageMax = NULL,
+                                            ageUnit = "year(s)",
+                                            BMIMin = NULL,
+                                            BMIMax = NULL,
+                                            BMIUnit = "kg/m²",
+                                            gestationalAgeMin = NULL,
+                                            gestationalAgeMax = NULL,
+                                            gestationalAgeUnit = "week(s)",
+
+                                            moleculeOntogenies = NULL) {
+
+  # Assuming that if this function is called directly, PKSim was either initialized already
+  # or should be initialized automatically
+  initPKSim()
+
+  validateIsString(species)
+  validateIsString(population, nullAllowed = TRUE)
+  validateIsNumeric(numberOfIndividuals)
+  validateIsNumeric(proportionOfFemales)
+
+  validateIsNumeric(weightMin, nullAllowed = TRUE)
+  validateIsNumeric(weightMax, nullAllowed = TRUE)
+  validateIsString(weightUnit)
+
+  validateIsNumeric(heightMin, nullAllowed = TRUE)
+  validateIsNumeric(heightMax, nullAllowed = TRUE)
+  validateIsString(heightUnit)
+
+  validateIsNumeric(ageMin, nullAllowed = TRUE)
+  validateIsNumeric(ageMax, nullAllowed = TRUE)
+  validateIsString(ageUnit)
+
+  validateIsNumeric(BMIMin, nullAllowed = TRUE)
+  validateIsNumeric(BMIMax, nullAllowed = TRUE)
+  validateIsString(BMIUnit)
+
+  validateIsNumeric(gestationalAgeMin, nullAllowed = TRUE)
+  validateIsNumeric(gestationalAgeMax, nullAllowed = TRUE)
+  validateIsString(gestationalAgeUnit)

```

Comment:
GAMin and GAMax must be an integer in [24..40] weeks. 
Also I would allow only "week (s)" as unit for GA.
Can we add this validations?

Reply:
Actually don'T know if GAMin=null or GAMax=null will work. In PK-Sim we set it explicitly to [24..40] if we want a full range.
In the Matlab implementation of GA we ask OSPSuiteCore for the allowed [Min..Max] Range of GA and set it to this range per default.
https://github.com/Open-Systems-Pharmacology/Matlab-Toolbox-Core/blob/970fcd890cc563c0656d605b092aa3da682959a3/code/auxiliaries/DefaultPopulationSettings.m#L20
https://github.com/Open-Systems-Pharmacology/Matlab-Toolbox-Core/blob/970fcd890cc563c0656d605b092aa3da682959a3/code/auxiliaries/DefaultPopulationSettings.m#L29-L30

Reply:
This uses the min max from the db if not specified. Nothing to change here in my opinion. 
We cannot check at the moment for integer values only

All units allowed for a given dimensions are allowed. So if there is something else than week(s) that's allowed. Why would we change this?